streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -111,6 +112,50 @@ abstract class DStream[T: ClassTag] (
   /* Set the creation call site */
   private[streaming] val creationSite = DStream.getCreationSite()
 
+  /**
+   * The base scope associated with the operation that created this DStream.
+   *
+   * This is the medium through which we pass the DStream operation name (e.g. updatedStateByKey)
+   * to the RDDs created by this DStream. Note that we never use this scope directly in RDDs.
+   * Instead, we instantiate a new scope during each call to `compute` based on this one.
+   *
+   * This is not defined if the DStream is created outside of one of the public DStream operations.
+   */
+  private[streaming] val baseScope: Option[String] = {
+    Option(ssc.sc.getLocalProperty(SparkContext.RDD_SCOPE_KEY))
+  }
+
+  /**
+   * An optional custom name for all scopes generated by this DStream.
+   * If None, the name of the operation that created this DStream will be used.
+   */
+  protected val customScopeName: Option[String] = None
```

Comment:
Do we not want users who create their own `DStreams` to be able to give their own names?